client/src/services/exchange.ts

```diff
@@ -0,0 +1,163 @@
+import { Office365GroupsActivityFileCounts, Office365GroupsActivityDetail } from '@microsoft/microsoft-graph-types-beta'
+import { graphAPIUrls } from '../graphHelper'
+import { makeGraphAPICall } from './base'
+import { UsersService } from './users'
+import { PeriodValueInDays } from '../types/general'
+
+export interface MailBoxUsageDetail {
+    reportRefreshDate: string
+    userPrincipalName: string
+    displayName: string
+    isDeleted: boolean
+    deletedDate: string | null
+    createdDate: string
+    lastActivityDate: string | null
+    itemCount: number
+    storageUsedInBytes: number
+    deletedItemCount: number
+    deletedItemSizeInBytes: number
+    deletedItemQuota: number
+    hasArchive: boolean
+    recipientType: string
+    issueWarningQuotaInBytes: number
+    prohibitSendQuotaInBytes: number
+    prohibitSendReceiveQuotaInBytes: number
+    reportPeriod: string
+}
+
+interface EmailActivityUserDetail {
+    reportRefreshDate: string
+    userPrincipalName: string
+    displayName: string
+    isDeleted: boolean
+    deletedDate: string | null
+    lastActivityDate: string
+    sendCount: number
+    receiveCount: number
+    readCount: number
+    meetingCreatedCount: number
+    meetingInteractedCount: number
+    assignedProducts: string[]
+    reportPeriod: string
+}
+
+interface EmailAppUserCount {
+    reportRefreshDate: string
+    mailForMac: null | number
+    outlookForMac: null | number
+    outlookForWindows: null | number
+    outlookForMobile: null | number
+    otherForMobile: null | number
+    outlookForWeb: number
+    pop3App: null | number
+    imap4App: null | number
+    smtpApp: null | number
+    reportPeriod: string
+}
+
+interface Storage {
+    reportRefreshDate: string
+    storageUsedInBytes: number
+    reportDate: string
+    reportPeriod: string
+}
+
+interface MailboxSettings {
+    '@odata.context': string
```

Comment:
Nitpicking: Technically if you are not using a props you don't have to include it (hence, interface)

Reply:
I believe these interfaces are for responses for API, hence i believe we can keep these here, for UI logic interfaces/types your suggestion totally makes sense, i think these types will help next developer working on same endpoint with same response, i can make a change if you suggest though :)